### PR TITLE
Redisdead v1.14

### PR DESCRIPTION
**Pull request by regit**
A rebase of #1507 with addressed comments.

PR builds:
- PR regit: https://buildbot.openinfosecfoundation.org/builders/regit/builds/105
- PR regit-pcap: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/103

### CODE REVIEW

File: src/util-logopenfile.c [hunk 1]
```diff
@@ -330,6 +330,138 @@ int SCConfLogReopen(LogFileCtx *log_ctx)
...
+            if (ret) {
+                log_ctx->redis_setup.batch_size = val;
+            } else {
+                log_ctx->redis_setup.batch_size = 10;
+            }
+        }
+    }
+
+    if (!strcmp(redis_mode, "list")) {
+        log_ctx->redis_setup.command = SCStrdup("LPUSH");
```
Comment (inliniac) [inline src/util-logopenfile.c]: why strdup here? Seems command is set in 2 places, both with hardcoded values. So could just make command a const char * instead and point to fixed strings.

File: src/util-logopenfile.c [hunk 2]
```diff
@@ -330,6 +330,138 @@ int SCConfLogReopen(LogFileCtx *log_ctx)
...
+
+    /* only try to reconnect once per second */
+    if (log_ctx->redis_setup.tried >= time(NULL)) {
+        return -1;
+    }
+
+    redisContext *c = redisConnect(log_ctx->redis_setup.server, log_ctx->redis_setup.port);
+    if (c != NULL && c->err) {
+        if (log_ctx->redis_setup.tried == 0) {
+            SCLogError(SC_ERR_SOCKET, "Error connecting to redis server: %s\n", c->errstr);
```
Comment (inliniac) [inline src/util-logopenfile.c]: newline at end of error msg

File: src/util-logopenfile.c [hunk 3]
```diff
@@ -375,9 +524,117 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
...
+                                file_ctx->redis->err);
+                    }
+                    switch (file_ctx->redis->err) {
+                        case REDIS_ERR_EOF:
+                        case REDIS_ERR_IO:
+                            SCLogInfo("Reopening connection to redis server");
+                            SCConfLogReopenRedis(file_ctx);
+                            if (file_ctx->redis) {
+                                SCLogInfo("Reconnected to redis server");
+                                SCMutexUnlock(&file_ctx->fp_mutex);
```
Comment (inliniac) [inline src/util-logopenfile.c]: this looks out of place, unlock happens unconditionally in LogFileWrite

File: src/util-logopenfile.c [hunk 4]
```diff
@@ -375,9 +524,117 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
...
+                                return 0;
+                            } else {
+                                SCLogInfo("Unable to reconnect to redis server");
+                                SCMutexUnlock(&file_ctx->fp_mutex);
```
Comment (inliniac) [inline src/util-logopenfile.c]: dito

File: src/util-logopenfile.c [hunk 5]
```diff
@@ -375,9 +524,117 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
...
+                                return 0;
+                            }
+                            break;
+                        default:
```
Comment (inliniac) [inline src/util-logopenfile.c]: Should this be an error/warning instead?
Also, should it lead to breaking the loop?

File: src/util-logopenfile.h [hunk 6]
```diff
@@ -34,13 +39,43 @@ typedef struct {
...
+
+typedef struct RedisSetup_ {
+    enum RedisMode mode;
+    char *command;
+    char *key;
+    int  batch_size;
+    int  batch_count;
+    char *server;
+    int  port;
+    int  tried;
```
Comment (inliniac) [inline src/util-logopenfile.h]: as this is set using time() the data type should be time_t

File: src/util-logopenfile.h [hunk 7]
```diff
@@ -34,13 +39,43 @@ typedef struct {
...
     union {
         FILE *fp;
         PcieFile *pcie_fp;
+#ifdef HAVE_LIBHIREDIS
+        redisContext *redis;
+#endif
+    };
+
+    union {
+    SyslogSetup syslog_setup;
```
Comment (inliniac) [inline src/util-logopenfile.h]: indent looks off in github

File: suricata.yaml.in [hunk 8]
```diff
@@ -100,6 +104,18 @@ outputs:
       #facility: local5
       #level: Info ## possible levels: Emergency, Alert, Critical,
                    ## Error, Warning, Notice, Info, Debug
+      #redis:
```
Comment (inliniac) [inline suricata.yaml.in]: Just to check: redis support would be automatically work for all log outputs ,right? Not in favour of adding this block everywhere of course.

Comment (regit) [inline suricata.yaml.in]: Yes, I've not tested but this should be the case.

File: src/util-logopenfile.h (hunk 7 above)
Comment (regit) [inline src/util-logopenfile.h]: Ack and fixed.

File: src/util-logopenfile.c (hunk 1 above)
Comment (regit) [inline src/util-logopenfile.c]: Ack and fixed.

File: src/util-logopenfile.c (hunk 2 above)
Comment (regit) [inline src/util-logopenfile.c]: Ack and fixed.

File: src/util-logopenfile.c (hunk 3 above)
Comment (regit) [inline src/util-logopenfile.c]: Outch. Fixed.

File: src/util-logopenfile.c (hunk 5 above)
Comment (regit) [inline src/util-logopenfile.c]: Yes, fixed.I was not sure on breaking the loop but that seems a sane decision.

File: src/util-logopenfile.h (hunk 6 above)
Comment (regit) [inline src/util-logopenfile.h]: Fixed.

**Pull request closed without merging**